doc/src/manual/environment-variables.md

```diff
@@ -156,14 +156,86 @@ The absolute path `REPL.find_hist_file()` of the REPL's history file. If
 $(DEPOT_PATH[1])/logs/repl_history.jl
 ```
 
+### `JULIA_MAX_NUM_PRECOMPILE_FILES`
+
+Sets the maximum number of different instances of a single package that are to be stored in the precompile cache (default = 10).
+
+## Pkg.jl
+
+### `JULIA_CI`
+
+If set to `true`, this indicates to the package server that any package operations are part of a continuous integration (CI) system for the purposes of gathering package usage statistics.
+
+### `JULIA_NUM_PRECOMPILE_TASKS`
+
+The number of parallel tasks to use when precompiling packages. See [`Pkg.precompile`](https://pkgdocs.julialang.org/v1/api/#Pkg.precompile).
+
+### `JULIA_PKG_DEVDIR`
+
+The default directory used by [`Pkg.develop`](https://pkgdocs.julialang.org/v1/api/#Pkg.develop) for downloading packages.
+
+### `JULIA_PKG_IGNORE_HASHES`
+
+If set to `1`, this will ignore incorrect hashes in artifacts. This should be used carefully, as it disables verification of downloads, but can resolve issues when moving files across different types of file systems. See [Pkg.jl issue #2317](https://github.com/JuliaLang/Pkg.jl/issues/2317) for more details.
+
+!!! compat "Julia 1.6"
+    This is only supported in Julia 1.6 and above.
+
+### `JULIA_PKG_OFFLINE`
+
+If set to `true`, this will enable offline mode: see [`Pkg.offline`](https://pkgdocs.julialang.org/v1/api/#Pkg.offline).
+
+!!! compat "Julia 1.5"
+    Pkg's offline mode requires Julia 1.5 or later.
+
+### `JULIA_PKG_PRECOMPILE_AUTO`
+
+If set to `0`, this will disable automatic precompilation by package actions which change the manifest. See [`Pkg.precompile`](https://pkgdocs.julialang.org/v1/api/#Pkg.precompile).
+
 ### `JULIA_PKG_SERVER`
 
-Used by `Pkg.jl`, for downloading packages and updating the registry. By default, `Pkg` uses `https://pkg.julialang.org` to
-fetch Julia packages. You can use this environment variable to select a different server. In addition, you can disable the use of the
-PkgServer protocol, and instead access the packages directly from their hosts (GitHub, GitLab, etc.) by setting:
-```
-export JULIA_PKG_SERVER=""
-```
+Specifies the URL of the package registry to use. By default, `Pkg` uses
+`https://pkg.julialang.org` to fetch Julia packages. In addition, you can disable the use of the PkgServer
+protocol, and instead access the packages directly from their hosts (GitHub, GitLab, etc.)
+by setting: ``` export JULIA_PKG_SERVER="" ```
+
+### `JULIA_PKG_SERVER_REGISTRY_PREFERENCE`
+
+Specifies the preferred registry flavor. Currently supported values are `conservative`
+(the default), which will only publish resources that have been processed, and `eager`,
+which will closely track the upstream repository, but may not have all resources
+available.
+
+!!! compat "Julia 1.7"
+    This only affects Julia 1.7 and above.
```

Comment:
@staticfloat I'm not sure I understand the difference, but I attempted to summarize it here, can you look it over?

Reply:
The difference is that `eager` starts serving the registry before the things included in that registry can be served by the package server, which means you can end up having to go to GitHub (or wherever) to get package versions and artifacts. The `complete` option waits until everything in the registry tree has been processed, so most of the time everything can be gotten from the package server. There are a few cases where a package server may still not have something referenced: if the package servers weren't able to download an artifact (or if it was too large > 1GB), then you'll still have to try to fetch the artifact directly. It should generally not happen that package versions aren't servable by the package server, but it could potentially happen.